src/ReleaseCommand.php

```diff
@@ -131,7 +132,28 @@ protected function execute(InputInterface $input, OutputInterface $output) : int
         $formatter = new ChangelogFormatter();
         $changelog = $formatter->format($changelog);
 
-        $remote = $input->getOption('remote') ?? 'origin';
+        $remotes = $this->fetchGitRemotes();
+        if (! $remotes) {
+            $output->writeln('<error>Cannot determine remote to which to push tag!</error>');
+            $output->writeln(
+                'The command "git remote -v" had a non-zero exit status; verify the command works, and try again.'
+            );
+            return 1;
+        }
+
+        $provider = $this->getProvider($config);
+        $remote   = $input->getOption('remote') ?: $this->lookupRemote(
```

Comment:
Possible improvement is to check if remote url matches expectation and to show notice message in the output in case it did not.